<launch>
  <arg name="other_topics" default=""/>
  <arg name="other_options" default="--split --size=10000"/>

  <!-- for BAXTER -->
  <arg name="save_openni" default="false" />
  <arg name="save_openni2" default="false" />
  <arg name="save_robot_model" default="false" />
  <arg name="save_json_data" default="true" />
  <arg name="save_object_verification" default="true" />
  <arg name="save_vision_sensors" default="true" />
  <arg name="save_baxter_model" default="true" />
  <arg name="save_all_image" default="true" />
  <arg name="save_all_button" default="false" />
  <arg name="save_hand_range" default="false" />
  <arg name="camera_namespace" default="kinect2"/>
  <arg name="save_dir" default="$(env HOME)/.ros/jsk_2015_05_baxter_apc/"/>
  <param name="mk_save_dir" command="mkdir -p $(arg save_dir)" />
  <!--                          -->
  <!--       Conditions         -->
  <!--                          -->

  <arg unless="$(arg save_all_button)" name="all_button_topics" value="" />
  <arg if="$(arg save_all_button)" name="all_button_topics" value=".*button.*/state" />

  <arg unless="$(arg save_hand_range)"
    name="hand_range_topics" value="" />
  <arg if="$(arg save_hand_range)" name="hand_range_topics"
    value="/robot/range/left_hand_range/state /robot/range/right_hand_range/state" />

  <!-- json data -->
  <arg unless="$(arg save_json_data)" name="json_data" value="" />
  <arg if="$(arg save_json_data)" name="json_data" value="/work_order/.* /bin_contents" />

  <!-- verification result -->
  <arg unless="$(arg save_object_verification)" name="object_verification" value=""/>
  <arg if="$(arg save_object_verification)" name="object_verification"
       value="/.*_process/bof_object_matcher/output /.*_process/object_verification/output"/>

  <arg unless="$(arg save_vision_sensors)" name="vision_sensors" value="" />
  <arg if="$(arg save_vision_sensors)" name="vision_sensors"
       value="/.*_process/apply_mask/output /.*_process/feature0d_to_image/output" />

  <!-- use original name space joint states -->
  <arg unless="$(arg save_baxter_model)" name="robot_model_topics" value="" />
  <arg if="$(arg save_baxter_model)" name="robot_model_topics" value="/tf /robot/joint_states" />

  <!-- execute -->
  <include file="$(find jsk_data)/launch/common_record.launch">
    <arg name="save_openni" value="$(arg save_openni)" />
    <arg name="save_openni2" value="$(arg save_openni2)" />
    <arg name="save_robot_model" value="$(arg save_robot_model)" />
    <arg name="save_all_image" value="$(arg save_all_image)" />
    <arg name="camera_namespace" value="$(arg camera_namespace)" />
    <arg name="save_dir" value="$(arg save_dir)" />
    <arg name="other_options" value="$(arg other_options)" />
    <arg name="other_topics"
      value="$(arg other_topics)
        $(arg robot_model_topics)
        $(arg hand_range_topics)" />
    <arg name="other_regex_topics"
      value="$(arg all_button_topics)
      $(arg json_data)
      $(arg vision_sensors)
      $(arg object_verification)"/>
  </include>
</launch>
